airflow/jobs.py

```diff
@@ -422,7 +424,7 @@ def schedule_dag(self, dag):
 
             # don't ever schedule prior to the dag's start_date
             if dag.start_date:
-                next_run_date = max(next_run_date, dag.start_date)
+                next_run_date = max(next_run_date or dag.start_date, dag.start_date)
```

Comment:
Nice catch. Perhaps a little more intelligible (though more verbose) would be:

    dag.start_date if not next_run_date else max(next_run_date, dag.start_date)